src/main/java/com/connexta/search/common/advice/CustomExceptionHandler.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2019 Connexta, LLC
+ *
+ * Released under the GNU Lesser General Public License version 3; see
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ */
+package com.connexta.search.common.advice;
+
+import com.connexta.search.common.exceptions.DetailedErrorAttributes;
+import java.util.Map;
+import javax.validation.ConstraintViolationException;
+import javax.validation.constraints.NotNull;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.RequestAttributes;
+import org.springframework.web.context.request.WebRequest;
+
+@ControllerAdvice
+@Slf4j
+@AllArgsConstructor
+public class CustomExceptionHandler {
+
+  @NotNull private final DetailedErrorAttributes detailedErrorAttributes;
+
+  @ExceptionHandler(ConstraintViolationException.class)
+  protected ResponseEntity<Object> handleConstraintViolation(
+      @NotNull final ConstraintViolationException e, @NotNull final WebRequest request) {
+    final HttpStatus status = HttpStatus.BAD_REQUEST;
+    log.warn("Request is invalid. Returning {}.", status, e);
+    final Map<String, Object> errorAttributes =
+        detailedErrorAttributes.getErrorAttributes(request, false);
+    errorAttributes.put("status", status.value());
+    errorAttributes.put("error", status.getReasonPhrase());
```

Comment:
❓ What will this `status.getReasonPhrase()` be since we just did a straight `final HttpStatus status = HttpStatus.BAD_REQUEST` right above?

Reply:
The value refers to the actual status code and the reason phrase would be `BAD REQUEST`

Reply:
Should we be cycling through the Constraint violations on the exception and grabbing those messages instead?

Reply:
Yeah, there's definitely room for improvement with what we send back in the response. This change was to just make sure we actually send back a formatted response.